charts/redpanda/chart_test.go

```diff
@@ -139,6 +140,26 @@ func TestTemplate(t *testing.T) {
 			})
 			require.NoError(t, err)
 
+			// kube-lint template file
+			var stdout bytes.Buffer
+			var stderr bytes.Buffer
+			inputYaml := bytes.NewBuffer(out)
+
+			cmd := exec.CommandContext(ctx, "kube-linter", "lint", "-", "--format", "json")
+			cmd.Stdin = inputYaml
+			cmd.Stdout = &stdout
+			cmd.Stderr = &stderr
+
+			errKubeLinter := cmd.Run()
```

Comment:
nit: in general, I prefer to use inline handling to avoid leaking errors into the larger scope.

```go
if err := ...; err != nil {
    // handle error here
}
```

That being said, I think the error handling here needs to be a bit weird. `kube-linter` could be erroring if it doesn't run correctly _or_ because it found errors. Given that we have the json output, we can pretty reasonably tell what's being reported:
```go
if errKubeLinter != nil && len(kubeLinterParsedJSONOutputErrorsField) > 0 {
 // Kube linter ran successfully but is reporting errors
} else if errKubeLinter != nil {
// Kube-linter did not run successfully.
} else {
// Kube-linter ran successfully and found no errors.
}

```